### PR TITLE
Wagtail 6.0

### PR DESCRIPTION
**Pull request by katdom13**
[Support Ticket](https://torchbox.monday.com/boards/1124794299/pulses/1396935580)

### Summary of changes

- Added Wagtail 6.0 compatibility
- Added tests for Django 5.0
- Added tests for Python 3.12

### CODE REVIEW

**Review by rachelhsmith — APPROVED** [review]
Hi @katdom13 this looks good to me!

File: wagtail_purge/wagtail_hooks.py [hunk 1]
```diff
@@ -14,9 +14,14 @@ def register_admin_urls():
 
 @hooks.register("register_settings_menu_item")
 def register_purge_menu_item():
+    if WAGTAIL_VERSION >= (5, 2):
+        kwargs = {"classname": "icon icon-collapse-down"}
+    else:
+        kwargs = {"classnames": "icon icon-collapse-down"}
+
```
Comment (zerolab) [inline wagtail_purge/wagtail_hooks.py]: This if statement is not needed since install requires `wagtail>=5.2`

Comment (katdom13) [reply wagtail_purge/wagtail_hooks.py]: Hi @zerolab ,
Good shout. Please see latest change:
https://github.com/torchbox/wagtail-purge/pull/10/commits/c4f063195e6ee0a9573749e456a11caee546a1f0

Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+
+- Support for Wagtail version 5.2 and above
```
Comment (zerolab) [inline CHANGELOG.md]:
```suggestion
- Updated supported Wagtail versions to 5.2 and above
```

File: wagtail_purge/wagtail_hooks.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from django.urls import include, reverse, path
 from wagtail.admin.menu import AdminOnlyMenuItem
 
-from wagtail import hooks
+from wagtail import hooks, VERSION as WAGTAIL_VERSION
```
Comment (zerolab) [inline wagtail_purge/wagtail_hooks.py]:
```suggestion
from wagtail import hooks
```

since `WAGTAIL_VERSION` is not used anywhere

**Review by zerolab — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (katdom13) [reply CHANGELOG.md]: Thanks @zerolab ,
Applied this suggestion

File: wagtail_purge/wagtail_hooks.py (hunk 3 above)
Comment (katdom13) [reply wagtail_purge/wagtail_hooks.py]: Applied this too

**Pull request merged**